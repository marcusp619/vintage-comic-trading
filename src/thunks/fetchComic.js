import md5 from "crypto-js/md5";
import { cleanComicData } from "../utils/DataCleaners";
import {
  isLoading,
  hasErrored,
  comicFetchDataSuccess,
  characterFetchDataSuccess
} from "../actions";
const PRIV_KEY = "05198a9099469d502c9138bd02417830becd2e8f";
const PUBLIC_KEY = "95d642f0dfe4844b8d9078f4e359a1af";

export const fetchCharacterData = () => {
  return async dispatch => {
    try {
      dispatch(isLoading(true));

      let timestamp = Date.now().toString();
      let hash = md5(`${timestamp}${PRIV_KEY}${PUBLIC_KEY}`);
      let url = `https://gateway.marvel.com:443/v1/public/characters?name=wolverine&apikey=${PUBLIC_KEY}&hash=${hash}`;
      const response = await fetch(url);
      dispatch(isLoading(false));
      const result = await response.json();
      cleanComicData(result);
      dispatch(characterFetchDataSuccess(result.data.results[0].name));
      return result.data.results[0];
    } catch {
      dispatch(isLoading(false));
      dispatch(hasErrored(true));
    }
  };
};

export const fetchComics = comicsUrlArray => {
  return dispatch => {
    dispatch(isLoading(true));
    const unresolvedPromises = comicsUrlArray.map(async comicUrl => {
      let timestamp = Date.now().toString();
      let hash = md5(`${timestamp}${PRIV_KEY}${PUBLIC_KEY}`);
      const url = `${comicUrl.resourceURI}`;
      console.log(url);
      // const response = await fetch(url);
      // const result = await response.json();
      // console.log(result);
      // return response;
    });
    // console.log(unresolvedPromises);
  };
};
