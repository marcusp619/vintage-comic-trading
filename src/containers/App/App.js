import React, { Component } from "react";
import { connect } from "react-redux";
import { fetchComics } from "../../thunks/fetchComic";
import Header from "../Header/";
import Hero from "../../components/Hero";
import styled from "styled-components";

export class App extends Component {
  async componentDidMount() {
    try {
      const character = await this.props.fetchComic();
    } catch (error) {
      throw Error(error.message);
    }
  }

  render() {
    return (
      <AppContainer>
        <Header />
        <Hero />
        <SectionWrapper>
          <SectionTitle>Your Comics</SectionTitle>
        </SectionWrapper>
      </AppContainer>
    );
  }
}

export const mapStateToProps = state => ({
  comics: state.comics,
  isLoading: state.loading,
  hasErrored: state.hasErrored
});

export const mapDispatchToProps = dispatch => {
  return {
    fetchComic: () => dispatch(fetchComics())
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);

const AppContainer = styled.div`
  background-color: #ccc7b9;
`;

const SectionWrapper = styled.section``;

const SectionTitle = styled.h2`
  color: #af7a6d;
  margin-top: 2em;
  text-align: center;
`;
