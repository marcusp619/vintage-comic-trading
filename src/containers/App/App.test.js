import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount } from "enzyme";

import { App, mapStateToProps, mapDispatchToProps } from "./App";
import { fetchComics } from "../../thunks/fetchComic";

describe("App Container", () => {
  it("should match snapshot", () => {
    //setup
    const wrapper = shallow(<App />);

    //expectation
    expect(wrapper).toMatchSnapshot();
  });

  describe("mapStateToProps", () => {
    it("should return an object to with the comics array", () => {
      //setup
      const mockState = {
        comics: [
          {
            id: 1009718,
            name: "Wolverine",
            description:
              "Born with super-human senses and the power to heal from almost any wound, Wolverine was captured by a secret Canadian organization and given an unbreakable skeleton and claws. Treated like an animal, it took years for him to control himself. Now, he's a premiere member of both the X-Men and the Avengers."
          }
        ],
        isLoading: undefined,
        hasErrored: undefined
      };
      const expected = {
        comics: [
          {
            id: 1009718,
            name: "Wolverine",
            description:
              "Born with super-human senses and the power to heal from almost any wound, Wolverine was captured by a secret Canadian organization and given an unbreakable skeleton and claws. Treated like an animal, it took years for him to control himself. Now, he's a premiere member of both the X-Men and the Avengers."
          }
        ],
        isLoading: undefined,
        hasErrored: undefined
      };

      //execution
      const mappedProps = mapStateToProps(mockState);

      //expectation
      expect(mappedProps).toEqual(expected);
    });
  });

  describe("mapDispatchToProps", () => {
    it("calls dispatch with an fetchComics action when fetchCharacterData is called", () => {
      //setup
      const mockDispatch = jest.fn();
      const actionToDispatch = fetchComics();

      //execution
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.fetchComics();

      //expectation
      expect(mockDispatch).toHaveBeenCalled();
    });
  });
});
