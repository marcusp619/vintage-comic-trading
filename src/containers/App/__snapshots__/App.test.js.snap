// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Container should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Connect(Header) />,
        <Hero />,
        <ForwardRef>
          <ForwardRef>
            Your Comics
          </ForwardRef>
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef>
            Your Comics
          </ForwardRef>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Your Comics",
          },
          "ref": null,
          "rendered": "Your Comics",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-dnqmqq",
              "isStatic": true,
              "rules": Array [
                "
  color: #af7a6d;
  margin-top: 2em;
  text-align: center;
",
              ],
            },
            "displayName": "styled.h2",
            "render": [Function],
            "styledComponentId": "sc-dnqmqq",
            "target": "h2",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-htoDjs",
            "isStatic": true,
            "rules": Array [],
          },
          "displayName": "styled.section",
          "render": [Function],
          "styledComponentId": "sc-htoDjs",
          "target": "section",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-gzVnrw",
        "isStatic": true,
        "rules": Array [
          "
  background-color: #ccc7b9;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-gzVnrw",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          <Hero />,
          <ForwardRef>
            <ForwardRef>
              Your Comics
            </ForwardRef>
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef>
              Your Comics
            </ForwardRef>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Your Comics",
            },
            "ref": null,
            "rendered": "Your Comics",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-dnqmqq",
                "isStatic": true,
                "rules": Array [
                  "
  color: #af7a6d;
  margin-top: 2em;
  text-align: center;
",
                ],
              },
              "displayName": "styled.h2",
              "render": [Function],
              "styledComponentId": "sc-dnqmqq",
              "target": "h2",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-htoDjs",
              "isStatic": true,
              "rules": Array [],
            },
            "displayName": "styled.section",
            "render": [Function],
            "styledComponentId": "sc-htoDjs",
            "target": "section",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-gzVnrw",
          "isStatic": true,
          "rules": Array [
            "
  background-color: #ccc7b9;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-gzVnrw",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
